js/viewer/shareoptions.js

```diff
@@ -0,0 +1,488 @@
+						cid: view.cid,
+						shareId: share.id,
+						shareWith: share.share_with,
+						name: shareOption.name,
+						label: shareOption.label,
+						enabled: shareOption.enabled
+					});
+				}
+
+				$share.append(
+					this._template({
+						shareOptions: shareOptionsData
+					})
+				);
+			}
+		}
+
+		// On click trigger logic to update for new richdocuments attributes
+		$(".richdocumentsShareOption").on('click', $.proxy(this.onOptionChange, this));
+	},
+
+	/**
+	 * Get attributes for the currently rendered checkboxes
+	 *
+	 * @param shareId
+	 * @returns {Array}
+	 * @private
+	 */
+	_getAttributesForShareOptions: function(shareId) {
+		var that = this;
+
+		var attributes = [];
+
+		// update attributes based on checkboxes
+		attributes = that._updateAttributes(attributes, "permissions", "download", null);
+		attributes = that._updateAttributes(attributes, "richdocuments", "watermark", null);
+		attributes = that._updateAttributes(attributes, "richdocuments", "print", null);
+		$(".richdocumentsShareOption").each(function(index, checkbox) {
+			if ($(checkbox).data('share-id') === shareId) {
+				var shareOptionName = $(checkbox).attr('name');
+				var shareOptionEnabled = $(checkbox).is(':checked');
+				if (shareOptionName === "secure-view" && shareOptionEnabled) {
+					attributes = that._updateAttributes(attributes, "permissions", "download", false);
+					attributes = that._updateAttributes(attributes, "richdocuments", "watermark", true);
+				} else if (shareOptionName === "secure-view" && !shareOptionEnabled) {
+					attributes = that._updateAttributes(attributes, "permissions", "download", true);
+					attributes = that._updateAttributes(attributes, "richdocuments", "watermark", false);
+				} else if (shareOptionName === "secure-view-print") {
+					attributes = that._updateAttributes(attributes, "richdocuments", "print", shareOptionEnabled);
+				}
+			}
+		});
+
+		return attributes;
+	},
+
+	/**
+	 * Get template share options for given share attributes
+	 *
+	 * @param attributes
+	 * @returns {array}
+	 * @private
+	 */
+	_getShareOptionsForAttributes: function(attributes) {
+		if (!attributes) {
+			return [];
+		}
+
+		var options = [];
+
+		// for each option, determine if share option is enabled based on required attributes
+		for(var optionIndex = 0; optionIndex < this._shareOptions.length; optionIndex++) {
+			var shareOption = this._shareOptions[optionIndex];
+
+			// get checkbox defined attributes
+			var checkboxAttributes = shareOption.attributes;
+
+			// determine if checkbox is enabled
+			// looping over checkbox defined and currently set attributes
+			var enabled = null;
+			for(var chIndex = 0; chIndex < checkboxAttributes.length; chIndex++) {
+				var checkboxAttribute = checkboxAttributes[chIndex];
+
+				for(var attrIndex = 0; attrIndex < attributes.length; attrIndex++) {
+					var currentAttribute = attributes[attrIndex];
+
+					if (currentAttribute.scope === checkboxAttribute.scope
```

Comment:
do we really need the "scope" property still or could a lot of these loops be implemented in a simpler way, for example by having the click handlers take care of exclusions in a hard-coded way ? (hard-coded would be fine as it's only 3 properties, if it helps with having less code and more readability)

Reply:
@PVince81 yes, scope is needed as this allows to resolve conflicts between apps on attributes. With app/core scope it is highly unlikely that someone writes 2 checkboxes for 2 different apps using the same 1 attribute. 

Regarding readability, this is true that it could just be hardcoded. Should I take care?

Reply:
I'm mostly considering that other apps might want to copy the code for this approach, and this is a lot of code. If you think that hard-coding more would significantly reduce the amount of code to copy, then go for it (separate PR?).

Otherwise let's leave this.